animations/lib/src/basics/06_tween_sequence.dart

```diff
@@ -0,0 +1,66 @@
+import 'package:flutter/material.dart';
+
+class TweenSequenceDemo extends StatefulWidget {
+  static const String routeName = '/basics/chaining_tweens';
+
+  @override
+  _TweenSequenceDemoState createState() => _TweenSequenceDemoState();
+}
+
+class _TweenSequenceDemoState extends State<TweenSequenceDemo>
+    with SingleTickerProviderStateMixin {
+  static const Duration duration = Duration(seconds: 1);
+  AnimationController controller;
+  Animation<Color> animation;
+
+  static final List<Color> colors = [
+    Colors.red,
+    Colors.orange,
+    Colors.yellow,
+    Colors.green,
+    Colors.blue,
+    Colors.indigo,
+    Colors.purple,
+  ];
+
+  void initState() {
+    super.initState();
+
+    final sequenceItems = <TweenSequenceItem<Color>>[];
+
+    for (var i = 0; i < colors.length; i++) {
+      final beginColor = colors[i];
+      final endColor = colors[(i + 1) % colors.length];
+      final weight = 1 / colors.length;
+
+      sequenceItems.add(
+        TweenSequenceItem<Color>(
+          tween: ColorTween(begin: beginColor, end: endColor),
+          weight: weight,
+        ),
+      );
+    }
+
+    controller = AnimationController(duration: duration, vsync: this);
+    animation = TweenSequence<Color>(sequenceItems).animate(controller)
+      ..addListener(() => setState(() {}));
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(),
+      body: Center(
+        child: MaterialButton(
+          color: animation.value,
+          onPressed: () {
+            controller.forward().then((_) {
```

Comment:
nit: use a param name rather than an underscore.

Reply:
In this case, the value is `void` since `controller.forward()` returns a `TickerFuture` (which is a `Future<void>`). And since it's unused, I would rather leave it an underscore. It's a convention of the language (https://github.com/dart-lang/site-www/issues/1101) IIRC but I don't see why the analyzer isn't giving a warning that it's unused when it's given a name (https://github.com/dart-lang/sdk/issues/23050)

Reply:
That's definitely a good reason to use the underscore. FWIW, I believe you could refactor the method to use async/await and get rid of it that way:

```dart
onPressed: () async {
  await controller.forward();
  controller.reset();
}
```

Reply:
I like that idea.  done